code/workspaces/infrastructure-api/src/kubernetes/deploymentApi.spec.js

```diff
@@ -141,26 +140,46 @@ describe('Kubernetes Deployment API', () => {
   describe('restart deployment', () => {
     it('should down and up scale deployment', async () => {
       // Arrange
-      const getSpy = jest.fn();
       const patchSpy = jest.fn();
-      mock.onGet(`${DEPLOYMENT_URL}/${DEPLOYMENT_NAME}/scale`).reply((requestConfig) => {
-        getSpy(requestConfig);
-        return [200, scale];
-      });
       mock.onPatch(`${DEPLOYMENT_URL}/${DEPLOYMENT_NAME}/scale`).reply((requestConfig) => {
         patchSpy(JSON.parse(requestConfig.data));
         return [204];
       });
+      const getSpy = jest.fn();
+      mock.onGet(`${DEPLOYMENT_URL}/${DEPLOYMENT_NAME}/scale`)
+        .replyOnce((requestConfig) => {
+          getSpy(requestConfig);
+          return [200, createScale(1, 1)]; // initial state
+        });
+      mock.onGet(`${DEPLOYMENT_URL}/${DEPLOYMENT_NAME}/scale`)
+        .replyOnce((requestConfig) => {
+          getSpy(requestConfig);
+          return [200, createScale(0, 1)]; // downscale incomplete
+        });
+      mock.onGet(`${DEPLOYMENT_URL}/${DEPLOYMENT_NAME}/scale`)
+        .replyOnce((requestConfig) => {
+          getSpy(requestConfig);
+          return [200, createScale(0, 0)]; // downscale complete
+        });
+      mock.onGet(`${DEPLOYMENT_URL}/${DEPLOYMENT_NAME}/scale`)
+        .replyOnce((requestConfig) => {
+          getSpy(requestConfig);
+          return [200, createScale(1, 0)]; // upscale incomplete
+        });
+      mock.onGet(`${DEPLOYMENT_URL}/${DEPLOYMENT_NAME}/scale`)
+        .replyOnce((requestConfig) => {
+          getSpy(requestConfig);
+          return [200, createScale(1, 1)]; // upscale complete
+        });
 
       // Act
-      const response = await deploymentApi.restartDeployment(DEPLOYMENT_NAME, NAMESPACE);
+      await deploymentApi.restartDeployment(DEPLOYMENT_NAME, NAMESPACE);
 
       // Assert
-      expect(getSpy).toHaveBeenCalledTimes(1);
+      expect(getSpy).toHaveBeenCalledTimes(5);
       expect(patchSpy).toHaveBeenCalledTimes(2);
       expect(patchSpy).toHaveBeenNthCalledWith(1, { spec: { replicas: 0 } });
       expect(patchSpy).toHaveBeenNthCalledWith(2, { spec: { replicas: 1 } });
-      expect(response.status).toEqual(204);
```

Comment:
Is there an indication from `restartDeployment` of whether is has succeeded or not? Is that something we are going to need?

Reply:
The key indication is that it hasn't thrown/rejected, but to make it consistent with other methods, `restartDeployment` now returns an axios response (the one for the patch which upscaled the deployment).

Reply:
Ahh I see. Thanks for the update Andy.